base/applications/taskmgr/endproc.c

```diff
@@ -107,6 +107,48 @@ BOOL IsCriticalProcess(HANDLE hProcess)
     return FALSE;
 }
 
+BOOL ShutdownProcessTreeHelper(HANDLE hSnapshot, HANDLE hParentProcess, DWORD dwParentPID)
+{
+    HANDLE hChildHandle;
+    PROCESSENTRY32W ProcessEntry = {0};
+    ProcessEntry.dwSize = sizeof(ProcessEntry);
+
+    if (Process32FirstW(hSnapshot, &ProcessEntry))
```

Comment:
Note: alternate implementation using NT API for finding parent process ID:
```c
NTSTATUS Status;
HANDLE ProcessId;
PROCESS_BASIC_INFORMATION ProcessInfo;
ULONG Length = sizeof(ProcessInfo);

/* Get the real parent's PID */

Status = NtQueryInformationProcess(TargetProcess->ProcessHandle,
                                   ProcessBasicInformation,
                                   &ProcessInfo,
                                   Length, &Length);
if (!NT_SUCCESS(Status))
{
    DPRINT1("Could not retrieve basic process info, Status = 0x%08lx\n", Status);
    return Status;
}

ProcessId = ULongToHandle(ProcessInfo.InheritedFromUniqueProcessId);
```

Reply:
We do not need to walk to the parent, we need to find all processes where our process is the parent.
aka when killing a process tree on notepad, we do not kill explorer (that started notepad) but we kill all processes started by notepad...

Reply:
I believe the current solution in my commit is good and there's no point in changing it to use the NT APIs (unless there's a good reason to do that).